syntax = "proto3";

package driverpacket;

message DriverPacket {
  repeated uint32 channels = 1;
  repeated SamplesBuffer channel_samples = 2;
  double rxrate = 3;
  uint32 sequence_num = 4;
  double txrate = 5;
  double txcenterfreq = 6;
  double rxcenterfreq = 7;
  uint32 numberofreceivesamples = 8;
  double timetosendsamples = 9;
  bool SOB = 10;
  bool EOB = 11;

  message SamplesBuffer {
    repeated double real = 1;
    repeated double imag = 2;
  }
}
